docs/tutorials/spec2006-tutorial-details.md

```diff
@@ -0,0 +1,636 @@
+        wget -O - https://gem5.googlesource.com/public/gem5-resources/+/a9db8cf1e2ea3c4b3ba84103afcdecfe345494c5/src/spec-2006/configs/system/fs_tools.py?format=TEXT | base64 --decode > fs_tools.py;
+        wget -O - https://gem5.googlesource.com/public/gem5-resources/+/a9db8cf1e2ea3c4b3ba84103afcdecfe345494c5/src/spec-2006/configs/system/system.py?format=TEXT | base64 --decode > system.py;
+    ''',
+    typ = 'git repo',
+    name = 'gem5-configs',
+    path =  'gem5-configs',
+    cwd = './',
+    documentation = 'gem5 run scripts made specifically for SPEC benchmarks'
+)
+```
+
+The gem5 run script, [run_spec.py](https://gem5.googlesource.com/public/gem5-resources/+/a9db8cf1e2ea3c4b3ba84103afcdecfe345494c5/src/spec-2006/configs/run_spec.py), takes the following parameters:
+* --kernel: (required) the path to vmlinux file.
+* --disk: (required) the path to spec image.
+* --cpu: (required) name of the detailed CPU model.
+Currently, we are supporting the following CPU models: kvm, o3, atomic, timing.
+More CPU models could be added to getDetailedCPUModel() in run_spec.py.
+* --benchmark: (required) name of the SPEC CPU 2006 benchmark.
+The availability of the benchmarks could be found at the end of the tutorial.
+* --size: (required) size of the benchmark. There are three options: ref, train, test.
+* --no-copy-logs: this is an optional parameter specifying whether the spec log files should be copied to the host system.
+* --allow-listeners: this is an optional parameter specifying whether gem5 should open ports so that gdb or telnet could connect to. No listeners are allowed by default.
+
+We don't use another Artifact object to document this file.
+The Artifact repository object of the root folder will keep track of the changes of the script.
+
+**Note:** The first two parameters of the gem5 run script for full system simulation should always be the path to the linux binary and the path to the disk image, in that order.
+
+## Run the Experiment
+### Setting up the Python virtual environment
+gem5art code works with Python 3.5 or above.
+
+The following script will set up a python3 virtual environment named gem5art-env. In the root folder of the experiment,
+
+```sh
+virtualenv -p python3 gem5art-env
+```
+
+To activate the virtual environment, in the root folder of the experiment,
+
+```sh
+source gem5art-env/bin/activate
+```
+
+To install the gem5art dependency (this should be done when we are in the virtual environment),
+
+```sh
+pip install gem5art-artifact gem5art-run gem5art-tasks
+```
+
+To exit the virtual environment,
+
+```sh
+deactivate
+```
+
+**Note:** the following steps should be done while using the Python virtual environment.
+
+### Running the Database Server
+The following script will run the MongoDB database server in a docker container.
+
```

Comment:
I'd refactor the following to have the script first, then a bullet point list explaining the flags/arguments.

Reply:
done